cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -85,17 +85,30 @@ type sfdiskPartition struct {
 	Name  string `json:"name"`
 }
 
-func (p *sfdiskPartition) isCreated() bool {
-	// TODO:UC20: also provide a mechanism for MBR (RPi)
-	if !creationSupported(p.Type) {
-		return false
-	}
-	for _, a := range strings.Fields(p.Attrs) {
-		if !strings.HasPrefix(a, "GUID:") {
-			continue
+func isCreatedDuringInstall(p *sfdiskPartition, fs *lsblkBlockDevice, sfdiskLabel string) bool {
+	switch sfdiskLabel {
+	case "gpt":
+		// the created partitions use specific GPT GUID types and set a
+		// specific bit in partition attributes
+		if !creationSupported(p.Type) {
+			return false
 		}
-		attrs := strings.Split(a[5:], ",")
-		if strutil.ListContains(attrs, createdPartitionAttr) {
+		for _, a := range strings.Fields(p.Attrs) {
+			if !strings.HasPrefix(a, "GUID:") {
+				continue
+			}
+			attrs := strings.Split(a[5:], ",")
+			if strutil.ListContains(attrs, createdPartitionAttr) {
+				return true
+			}
+		}
+	case "dos":
+		// we have no similar type/bit attribute setting for MBR, on top
+		// of that MBR does not support partition names, fall back to
+		// reasonable assumption that partitions carrying a filesystem
+		// labeled "ubuntu-seed" are part of the factory image while
+		// everything else was created at install time
+		if fs.Label != ubuntuSeedLabel {
```

Comment:
I wonder if we should be slightly more explicit here and realy encode ["ubuntu-data", "ubuntu-boot", "ubuntu-save"] because the gadget may have partitions for firmware or in the LK kernel case for storing raw kernels etc?

Reply:
Fair point. I'll push it.

Reply:
yes, we definitely need to be explicit here (that's what I suggested originally I think, though I forgot that mbr entries themselves carry no names :/ )

Reply:
This will cover the factory reset use case, but if we crash during install we may still need to reflash the image (because it could crash before the filesystem is created). But this is not a very likely scenario and reflashing is easy.

Reply:
maybe we should add a TODO about considering a mapping from start offset to label (via the gadget info) ?